sdks/python/apache_beam/utils/pipeline_options.py

```diff
@@ -20,6 +20,69 @@
 import argparse
 
 from apache_beam.transforms.display import HasDisplayData
+from apache_beam.utils.value_provider import StaticValueProvider
+from apache_beam.utils.value_provider import RuntimeValueProvider
+from apache_beam.utils.value_provider import ValueProvider
+
+
+def static_value_provider_of(value_type):
+  """"Helper function to plug a ValueProvider into argparse.
+
+  Args:
+    value_type: the type of the value. Since the type param of argparse's
+                add_argument will always be ValueProvider, we need to
```

Comment:
Yes, it is. What I mean is to say why add_argument would fall short (because it wouldn't preserve the specific type of the value provider) so this value_type is needed.